### PR TITLE
Validate PING, DNS and TCP targets

### PR DESCRIPTION
**Pull request by mem**
Add validations for the targets of PING, DNS and TCP checks. Make sure
we are getting something that is either a valid IP address or a valid
hostname. The API will also perform this validation, possibly more
strictly, so this is only meant to give the user some immediate
feedback.

Signed-off-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>

### CODE REVIEW

File: src/validation.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { Check, CheckType, Settings, HttpSettings, PingSettings, DnsSettings, TcpSettings, Label } from 'types';
 import { checkType } from 'utils';
+const punycode = require('punycode');
+const ipaddress = require('ip-address');
```
Comment (ryantxu) [inline src/validation.ts:4]: can we import rather than require?

File: src/validation.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { validateCheck } from 'validation';
+import { Check } from 'types';
+
+describe('validation test', () => {
+  it('should test target', () => {
+    const check: Check = {
```
Comment (ryantxu) [inline src/validation.test.ts:6]: @mem -- can you add more tests here?  that will help whoever has to look at this next :)

File: src/validation.ts (hunk 1 above)
Comment (mem) [reply src/validation.ts]: thanks!

File: src/validation.test.ts (hunk 2 above)
Comment (mem) [reply src/validation.test.ts:6]: I added a bunch, thanks for bootstrapping! :smile:

**Pull request merged**